lib/vrl/compiler/src/expression/function_call.rs

```diff
@@ -196,9 +202,62 @@ impl FunctionCall {
             arguments_fmt,
             arguments_dbg,
             ident: function.identifier(),
+            function_id,
+            arguments: Arc::new(arguments),
         })
     }
 
+    /// Takes the arguments passed and resolves them into the order they are defined
+    /// in the function
+    /// The error path in this function should never really be hit as the compiler should
+    /// catch these whilst creating the AST.
```

Comment:
Perhaps in the future. However I have found that by returning `Result` it works much better with the VRL tests. ie.. we get a properly printed diagnostic error together with the test that failed.

With a panic it just dumps the panic, doesn't say which test failed, and stops all ensuing tests. 

Because of this I would like to avoid panics where possible.